contracts/gauges/LiquidityGaugeV5Collateral.vy

```diff
@@ -515,17 +522,26 @@ def _transfer(_from: address, _to: address, _value: uint256):
     if _value != 0:
         total_supply: uint256 = self.totalSupply
         is_rewards: bool = self.reward_count != 0
+        is_whitelisted_custody_from: bool = self.whitelisted_custody[_from]
+        is_whitelisted_custody_to: bool = self.whitelisted_custody[_to]
+
         if is_rewards:
             self._checkpoint_rewards(_from, total_supply, False, ZERO_ADDRESS)
         new_balance: uint256 = self.balanceOf[_from] - _value
         self.balanceOf[_from] = new_balance
-        self._update_liquidity_limit(_from, new_balance, total_supply)
+        if(!is_whitelisted_custody_from and !is_whitelisted_custody_to):
```

Comment:
Scenarios:

1. user -> lending pool. In this case we do not want to change the adjusted balance of the **user** (from) and **lending pool** (to) because they should still be accumulating rewards while they are lending in pool.
2. lending pool -> user. In this case we do not want to change the adjusted balance of the **lending pool** (from), but we do want to change the adjusted balance of the **user** (to) because they received tokens. 
3. user -> user. In this case we want to change the adjusted balance of **user** (from) and **user** (to).

Reply:
Is this line transfer scenario 3?  user -> user?  I'm having a difficult time understanding where each transfer scenario is trigger.

Reply:
> Is this line transfer scenario 3? user -> user? I'm having a difficult time understanding where each transfer scenario is trigger.

yes that line is triggered on scenario 3

1. `!false and !true` (false)
2. `!true and !false` (false)
3. `!false and !false` (true)

Reply:
Could we abstract each transfer scenario out into its own function? It would be easier to unit test and understand the logic.